Patch to add an 'install' target for the UNIX Makefile, to allow the program to be installed.
Author: John Robert Beshir
Licence: Patch is public domain; original program is proprietary.
Contributed Upstream: Submitted.
--- unrar/makefile.unix	2005-04-30 07:00:12.000000000 +0000
+++ unrar/makefile.unix.new	2008-05-31 02:25:02.000000000 +0000
@@ -10,30 +10,35 @@
 CXXFLAGS=-O2
 DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 STRIP=strip
+DESTDIR=/usr
 
 # Linux using LCC
 #CXX=lcc
 #CXXFLAGS=-O2
 #DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 #STRIP=strip
+#DESTDIR=/usr
 
 # HP UX using aCC
 #CXX=aCC
 #CXXFLAGS=-AA +O2 +Onolimit
 #DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 #STRIP=strip
+#DESTDIR=/usr
 
 # IRIX using GCC
 #CXX=g++
 #CXXFLAGS=-O2 
 #DEFINES=-DBIG_ENDIAN -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_BSD_COMPAT -D_XOPEN_SOURCE -D_XOPEN_SOURCE_EXTENDED=1
 #STRIP=strip
+#DESTDIR=/usr
 
 # IRIX using MIPSPro (experimental)
 #CXX=CC
 #CXXFLAGS=-O2 -mips3 -woff 1234,1156,3284 -LANG:std
 #DEFINES=-DBIG_ENDIAN -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_BSD_COMPAT -DNATIVE_INT64 -DInt64=int64_t
 #STRIP=strip
+#DESTDIR=/usr
 
 # AIX using xlC (IBM VisualAge C++ 5.0)
 #CXX=xlC
@@ -41,36 +46,42 @@
 #DEFINES=-D_LARGE_FILES -D_LARGE_FILE_API
 #LIBS=-lbsd
 #STRIP=strip
+#DESTDIR=/usr
 
 # Solaris using CC (SUN Forte Developer 7 C++)
 #CXX=CC
 #CXXFLAGS=-xO2 -xbuiltin=%all -xinline=%auto
 #DEFINES=-DBIG_ENDIAN -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 #STRIP=strip
+#DESTDIR=/usr
 
 # Solaris using GCC (optimized for UltraSPARC 1 CPU)
 #CXX=g++
 #CXXFLAGS=-O3 -mcpu=v9 -mtune=ultrasparc -m32
 #DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 #STRIP=/usr/ccs/bin/strip
+#DESTDIR=/usr
 
 # Tru64 5.1B using GCC3
 #CXX=g++
 #CXXFLAGS=-O2 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_XOPEN_SOURCE=500
 #STRIP=strip
 #LDFLAGS=-rpath /usr/local/gcc/lib
+#DESTDIR=/usr
 
 # Tru64 5.1B using DEC C++
 #CXX=cxx
 #CXXFLAGS=-O4 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DNATIVE_INT64 -DInt64=long
 #STRIP=strip
 #LDFLAGS=
+#DESTDIR=/usr
 
 # QNX 6.x using GCC
 #CXX=g++
 #CXXFLAGS=-O2 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -fexceptions
 #STRIP=strip
 #LDFLAGS=-fexceptions
+#DESTDIR=/usr
 
 # Cross-compile
 # Linux using arm-linux-g++
@@ -79,6 +90,7 @@
 #DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 #STRIP=arm-linux-strip
 #LDFLAGS=-static
+#DESTDIR=/usr
 
 ##########################
 
@@ -98,6 +110,10 @@
 
 all:	unrar
 
+install:	install-unrar
+
+uninstall:	uninstall-unrar
+
 clean:
 	@rm -f *.o *.bak *~
 
@@ -117,3 +133,15 @@
 lib:	$(OBJECTS) $(LIB_OBJ)
 	@rm -f libunrar.so
 	$(LINK) -shared -o libunrar.so $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
+
+install-unrar:
+			install unrar $(DESTDIR)/bin
+
+uninstall-unrar:
+			rm -f $(DESTDIR)/bin/unrar
+
+install-lib:
+		install libunrar.so $(DESTDIR)/lib
+
+uninstall-lib:
+		rm -f $(DESTDIR)/lib/libunrar.so
